src/core/main.js

```diff
@@ -231,6 +231,12 @@ class p5 {
       this._events.devicemotion = null;
     }
 
+    // before and after hooks for preload and setup
+    this._beforePreload = function () {};
```

Comment:
We should probably loop through and call everything in `target._registeredMethods[key]` for each of these

Reply:
With the method above, how do we register a method to, say, afterSetup? 
Would this work?
```p5.on("afterSetup", function(){...})```

Reply:
Pretty close! It should be `p5.registerMethod('afterSetup', function() { ... })` (and `unregisterMethod` to remove it later)